react/components/MainContainer.tsx

```diff
@@ -1,21 +1,33 @@
 import React, { useContext, useEffect, useCallback } from 'react'
 import { Box, Skeleton, Center } from '@vtex/admin-ui'
+import { useIntl } from 'react-intl'
+import { showToast } from 'vtex.admin-shell-utils'
 
 import TikTokContext from '../contexts/TikTokContext'
 import StartupForm from './StartupForm'
 import IntegrationOptionsProvider from '../providers/IntegrationOptionsProvider'
 import { FORM_HEIGHT } from '../utils/constants'
+import { MESSAGES } from '../utils/messages'
 
 const MainContainer = () => {
   const context = useContext(TikTokContext)
+  const intl = useIntl()
 
   const refetchSettings = useCallback(
     (event) => {
       if (event.origin === window.location.origin) {
+        if (event.data === 'access-token-error') {
```

Comment:
This `if` can be used with a aditional condition inside the above `if`, can't?

```suggestion
      if (event.origin === window.location.origin && event.data === 'access-token-error') {
```

Reply:
Yes it can, but the `widown.location.origin` always need to be verified for any action. I can do it to get rid of the `if` nesting, but i will need another `if` for the second case